bodo/utils/py_objs.py

```diff
@@ -122,3 +127,73 @@ def create_struct_from_pyobject(typ, obj, context, builder, pyapi):
     )
     struct_proxy.pyobj = obj
     return struct_proxy._getvalue()
+
+
+def install_opaque_class(
```

Comment:
This is potentially dangerous because it doesn't delete the object after use. Can you confirm that memory management is done manually for all the uses properly?

Reply:
So I think i reworked it multiple times to the point where only `python_list_of_heterogeneous_tuples_type` needs to be opaque now, and I'm certain we deallocate it correctly in C++. I modified the others to use `install_py_obj_class`.

Reply:
Oh my bad, also `pyarrow_fs_type` uses it, but the `catalog` object created and owns it. Since the catalog object is still around as long as the transaction is, it should be fine to not delete it in C++.